tools/smartos-release

```diff
@@ -78,31 +82,40 @@ set -o errexit
 # Note that ${BRANCH} appears only once here as we assume that for release
 # builds, all project branches and smartos-live itself use the same branch
 # name. See $(PUB_BRANCH_DESC) in the top-level Makefile for more details.
-SOURCE=/${MANTA_USER}/public/builds/platform/${BRANCH}-${TIMESTAMP}/platform/
+SOURCE=/${MANTA_USER}/public/builds/platform/${BRANCH}-${TIMESTAMP}/platform
 SMARTOS=/${MANTA_USER}/public/SmartOS
 DESTINATION=${SMARTOS}/${TIMESTAMP}
 
 start_time=$(date +%s)
-echo "Creating release snaplinks under ${DESTINATION}"
+echo "Creating release objects under ${DESTINATION}"
 
 mmkdir -v -p ${DESTINATION}
-mfind ${SOURCE} -t o | while read OBJECT; do
-    mln ${OBJECT} ${DESTINATION}/$(basename ${OBJECT})
-done
-
-echo "Updating top level ${SMARTOS} snaplinks"
-mln ${SOURCE}/platform-${BRANCH}-${TIMESTAMP}.tgz ${SMARTOS}/platform-latest.tgz
-mln ${SOURCE}/smartos-${TIMESTAMP}.iso ${SMARTOS}/smartos-latest.iso
-mln ${SOURCE}/smartos-${TIMESTAMP}-USB.img.gz ${SMARTOS}/smartos-latest-USB.img.gz
-mln ${SOURCE}/smartos-${TIMESTAMP}.vmwarevm.tar.gz ${SMARTOS}/smartos-latest.vmwarevm.tar.gz
+# Add an object pointing to the actual platform Manta directory, along with
+# a html version, for convenience.
+echo ${SOURCE} | mput ${DESTINATION}/platform-link
+echo "
+<html>
+<head><title>SmartOS ${BRANCH}-${TIMESTAMP}</title></head>
+<body>
+<a href='../../builds/platform/${BRANCH}-${TIMESTAMP}/platform/index.html'>
+$BRANCH-$TIMESTAMP artifacts</a>
+</body>
+</html>" | mput -H 'content-type: text/html' ${DESTINATION}/platform-link.html
+
+echo "Uploading 'latest' ${SMARTOS} objects to ${SMARTOS}"
+BITS=output/bits/platform
+mput -f $BITS/platform-${TIMESTAMP}.tgz ${SMARTOS}/platform-latest.tgz
+mput -f $BITS/output/smartos-${TIMESTAMP}.iso ${SMARTOS}/smartos-latest.iso
+mput -f $BITS/smartos-${TIMESTAMP}-USB.img.gz ${SMARTOS}/smartos-latest-USB.img.gz
+mput -f $BITS/smartos-${TIMESTAMP}.vmwarevm.tar.gz ${SMARTOS}/smartos-latest.vmwarevm.tar.gz
 
 echo "Updating ${SMARTOS}/latest object"
-echo ${DESTINATION} | mput -v -H 'content-type: text/plain' ${SMARTOS}/latest
+echo ${SOURCE} | mput -v -H 'content-type: text/plain' ${SMARTOS}/latest
```

Comment:
I'm not following this change

Reply:
$SOURCE is the real Manta location for the platform artifacts (/builds/platform/${BRANCH}-${TIMESTAMP}/platform/) rather than SmartOS directory which only contained a bunch of snaplinks. Since we're removing the reliance on snaplinks, we should have 'latest' point to the directory that contains the actual objects.